tools/integration_tests_runner.py

```diff
@@ -2,136 +2,130 @@
 from time import time
-import traceback
-
 
 from run_desktop_tests import tests_on_disk
 
 __author__ = 't.danshin'
 
 
+MAIN_LOGGER_NAME = "main_logger"
+START_FINISH_LOGGER_NAME = "start_finnish_logger"
 TEMPFOLDER_TESTS = ["search_integration_tests", "storage_integration_tests"]
 
 
-class IntegrationRunner:
-    def __init__(self):
-        self.process_cli()
+def setup_logger(logger_name, log_file, level=logging.INFO):
+    l = logging.getLogger(logger_name)
+    formatter = logging.Formatter('%(msg)s')
+    file_handler = logging.FileHandler(log_file, mode='w')
+    file_handler.setFormatter(formatter)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
 
-        self.proc_count = multiprocessing.cpu_count()
-        logging.info("Number of processors is: {nproc}".format(nproc=self.proc_count))
+    l.setLevel(level)
+    l.addHandler(file_handler)
+    l.addHandler(stream_handler)
 
-        self.file_lock = Lock()
-        self.start_finish_lock = Lock()
-        self.tests = Queue()
-        self.key_postfix = ""
-        if self.user_resource_path:
-            self.key_postfix += ' --user_resource_path="{0}"'.format(self.user_resource_path)
-        if self.data_path:
-            self.key_postfix += ' --data_path="{0}"'.format(self.data_path)
 
+def execute_file(test, keys):
+    spell = ["{test} {keys}".format(test=test, keys=keys)]
+    logging.debug(">> {spell}".format(spell=spell))
 
-    def run_tests(self):
-        intermediate_tests = []
-        for exec_file in self.runlist:
-            intermediate_tests.extend(map(lambda x: (exec_file, x), self.get_tests_from_exec_file(exec_file, "--list_tests")[0]))
+    process = subprocess.Popen(spell,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               shell=True
+                               )
 
-        shuffle(intermediate_tests)
-        for test in intermediate_tests:
-            self.tests.put(test)
+    out, err = process.communicate()
+    result = process.returncode
 
-        with open(self.output, "w") as self.file, open("start-finish.log", "w") as self.start_finish_log:
-            self.run_parallel_tests()
+    return filter(None, out.split("\n")), err, result
 
 
-    def run_parallel_tests(self):
-        threads = []
+def exec_test(a_tuple):
+    test_file, test, key_postfix = a_tuple
+    keys = '"--filter={test}"'.format(test=test)
 
-        for i in range(0, self.proc_count):
-            thread = Thread(target=self.exec_tests_in_queue)
-            thread.start()
-            threads.append(thread)
+    if test_file in TEMPFOLDER_TESTS:
+        tmpdir = tempfile.mkdtemp()
+        keys = '{old_key} "--data_path={tmpdir}"'.format(old_key=keys, tmpdir=tmpdir)
+        logging.debug("Temp dir: {tmpdir}".format(tmpdir=tmpdir))
+    else:
+        keys = "{old_key}{resource_path}".format(old_key=keys, resource_path=key_postfix)
```

Comment:
I do not understand this line? What is the point to append (without any space and without flag specification) resource path to the |keys| variable?